gradle/dependency-management/capabilities.json

```diff
@@ -7,6 +7,17 @@
 // the same capability, and that we prefer to use slf4j in this case.
 [
     // Soft upgrades: only in case of conflict, will use the latest version of the selected module in the graph
+    {
+        // hamcrest packaging changes, see https://hamcrest.org/JavaHamcrest/distributables#previous-versions-of-hamcrest
+        "name": "hamcrest",
+        "providedBy": [
+            "org.hamcrest:hamcrest",
```

Comment:
The `selected` entry should not be included in the `providedBy` list -- notice the other entires in this file also do not do this

See the [logic](https://github.com/gradle/gradle/blob/a00d0067d03d635d09086b360a922b0175e17c79/build-logic/dependency-modules/src/main/kotlin/gradlebuild.dependency-modules.gradle.kts#L139-L166), which adds a `CapabilityRule` for all `providedBy` entries and the `selected` entry

Reply:
I think the groovy item is what inspired this, https://github.com/gradle/gradle/blob/61cfead7afdae186c1770058f309f712a4b85c4e/gradle/dependency-management/capabilities.json#L47-L54

from a quick search it looks like it has no effect any more and could probably be removed...